### DISCUSSION TITLE
Upgreade flux v2 version

### DISCUSSION THREAD

**Discussion by eravindar12** [fluxcd/flux2#2840] [Q&A] [answered]
Hi All,

I'd set it up below the flux version (i.e., including all components)and not want to upgrade it to the latest version 
Can you please guide me on how to do that?

flux version
flux: v0.28.5
helm-controller: v0.18.2
image-automation-controller: v0.21.3
image-reflector-controller: v0.17.1
kustomize-controller: v0.22.3
notification-controller: v0.23.2
source-controller: v0.22.5

**Comment by kingdonb** [comment, marked as answer]
If you install a Flux binary from the releases listing: https://github.com/fluxcd/flux2/releases at a particular version, then it will install the components from that version. Flux CLI will in some cases (eg. `flux check`) warn you that you are installing from a not-latest version, but it will still do so.

The particular version you selected is one just before our last round of CVE fix announcements:

https://github.com/fluxcd/flux2/security/advisories/GHSA-7pwf-jg34-hxwp
https://github.com/fluxcd/flux2/security/advisories/GHSA-j77r-2fxf-5jrw
https://github.com/fluxcd/flux2/security/advisories/GHSA-vvmq-fwmg-2gjc

I would recommend installing a later version, unless you are doing this for security research 👍
